src/app/dim-ui/usePopper.ts

```diff
@@ -160,5 +163,16 @@ export function usePopper({
     }
 
     return destroy;
-  });
+  }, [
+    contents,
+    reference,
+    arrowClassName,
+    menuClassName,
+    boundarySelector,
+    placement,
+    offset,
+    fixed,
+    padding,
+    deps,
```

Comment:
React expects that all reactive values be listed as dependencies: https://react.dev/reference/react/useEffect#specifying-reactive-dependencies

Reply:
Fair, though in practice many of these are immutable. Can't hurt, though.

What doesn't work is including `deps` directly like this - hooks' dependencies list won't recursively evaluate equality. So you probably want:
```suggestion
    contents,
    reference,
    arrowClassName,
    menuClassName,
    boundarySelector,
    placement,
    offset,
    fixed,
    padding,
    ...deps,
```

Reply:
Ah yeah, so that causes a React warning since React can't statically analyze the dependencies that way. We can add a comment to ignore the rule, but wasn't sure if you'd be comfortable with that approach.

-----

I can also remove the other refs if they are immutable, though that will cause React to log another warning since it expects them to be included as dependencies. Similarly, we can add a comment to ignore the rule if you'd like.

Reply:
Probably the best would be to silence the warning about having a dynamic dependency list, but still include all the other dependencies.

Reply:
All set!